### PR TITLE
Fix variable on restart and remove old webview code path

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #7970

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [x] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/datascience-ui/interactive-common/redux/reducers/variables.ts [hunk 1]
```diff
@@ -238,8 +238,8 @@ function handleRestarted(arg: VariableReducerArg): IVariableState {
                 sortColumn: arg.prevState.sortColumn,
                 sortAscending: arg.prevState.sortAscending,
                 startIndex: 0,
-                pageSize: arg.prevState.pageSize,
-                refreshCount: 0
+                pageSize: 5,
+                refreshCount: arg.prevState.refreshCount
```
Comment (IanMatthewHuff) [inline src/datascience-ui/interactive-common/redux/reducers/variables.ts:242]: This here is the actual fix, when we moved from hosting this in each webview view to hosting a single instance in the variable view we changed how we use refreshCount some. If you switch between documents you can sometime have the same execution count as before, in which case we bump up the refresh count to force a refresh. 

However on restart this could lead to a state where we could have the same execution count and the same refresh count. In this case it would not realized that it needed to fetch new data, and you could get odd places where you would get Loading... and the rows would never load. Just keep refreshCount always counting up even after a restart to prevent this.

**Review by rchiodo — APPROVED** [review]

File: src/datascience-ui/interactive-common/variableExplorer.tsx [hunk 2]
```diff
@@ -182,31 +176,19 @@ export class VariableExplorer extends React.Component<IVariableExplorerProps, IV
         if (!fastDeepEqual(this.props.variables, nextProps.variables)) {
             return true;
         }
-        if (
-            prevState.containerHeight !== this.state.containerHeight ||
-            (prevState.gridHeight !== this.state.gridHeight && !this.props.standaloneMode)
-        ) {
+        if (prevState.containerHeight !== this.state.containerHeight) {
```
Comment (IanMatthewHuff) [inline src/datascience-ui/interactive-common/variableExplorer.tsx:179]: The real fix was above, the other changes here (maybe should have done this in a separate PR, but I was cleaning it up while working so I figured why not) are removing the few special cases used for when we hosted this control in a webview document versus the new single webview view. We now only have the webview view case.
